src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -223,6 +223,7 @@ public function has($id)
             if ('service_container' === $id
                 || isset($this->aliases[$id])
                 || isset($this->services[$id])
```

Comment:
container->has/get() is full of micro-optimization because it can be called thousands of time.
Which immediately unqualifies this as a "micro"-optimization and make it a "macro" one.